lib/ansible/modules/cloud/amazon/lightsail.py

```diff
@@ -0,0 +1,405 @@
+
+    # Wait for instance to exit transition state before state change
+    if wait:
+        while wait_max > time.time() and inst is not None and inst['state']['name'] in ('pending', 'stopping'):
+            try:
+                time.sleep(5)
+                inst = find_instance_info(client, instance_name)
+            except botocore.exceptions.ClientError as e:
+                time.sleep(1)
+
+    # Try state change
+    if inst is not None and inst['state']['name'] != state:
+        try:
+            if state == 'running':
+                client.start_instance(instanceName=instance_name)
+            else:
+                client.stop_instance(instanceName=instance_name)
+        except botocore.exceptions.ClientError as e:
+            module.fail_json(msg='Unable to change state for instance {0}, error: {1}'.format(instance_name, e))
+        changed = True
+        # Grab current instance info
+        inst = find_instance_info(client, instance_name)
+
+    return (changed, inst)
+
+def core(module):
+    region, ec2_url, aws_connect_kwargs = get_aws_connection_info(module, boto3=True)
+    if not region:
+        module.fail_json(msg='region must be specified')
+
+    client = None
+    try:
+        client = boto3_conn(module, conn_type='client', resource='lightsail',
+                            region=region, endpoint=ec2_url, **aws_connect_kwargs)
+    except (botocore.exceptions.ClientError, botocore.exceptions.ValidationError) as e:
+        module.fail_json(msg=str(e))
+
+    changed = False
+    state = module.params['state']
+    name = module.params['name']
+
+    if state == 'absent':
+        (changed, instance_dict) = delete_instance(module, client, name)
+    elif state in ('running', 'stopped'):
+        (changed, instance_dict) = startstop_instance(module, client, name, state)
+    elif state == 'restarted':
+        (changed, instance_dict) = restart_instance(module, client, name)
+    elif state == 'present':
+        (changed, instance_dict) = create_instance(module, client, name)
+
+    module.exit_json(changed=changed, instance=instance_dict)
+
+def find_instance_info(client, instance_name):
+    inst = None
+    try:
+        inst = client.get_instance(instanceName=instance_name)
+    except botocore.exceptions.ClientError as e:
+        raise
+    return inst['instance']
+
+def main():
+    argument_spec = ec2_argument_spec()
+    argument_spec.update(dict(
+        name=dict(type='str', required=True),
+        state=dict(type='str', default='present', choices=['present', 'absent', 'stopped', 'running', 'restarted']),
+        zone=dict(type='str'),
+        blueprint_id=dict(type='str'),
+        bundle_id=dict(type='str'),
+        key_pair_name=dict(type='str'),
+        user_data=dict(type='str'),
+        wait=dict(type='bool', default=True),
+        wait_timeout=dict(default=300),
+    ))
+
+    module = AnsibleModule(argument_spec=argument_spec)
+
+    if not HAS_BOTOCORE:
```

Comment:
Botocore is a dependency of boto3, but this belt-and-suspenders approach of checking both is ok. Can you swap the order so that a user will be prompted to install boto3, which would also resolve a lack of botocore? Basically just swap this conditional with lines 394/395.